### PR TITLE
Implement commands for team management

### LINKED ISSUE #8918: Implement command for teams

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #8918

### CODE REVIEW

**Review by jeremyletang — COMMENTED** [review]
there should be another oneoff to add. these, in the events proto package, something like TransactionResult.

File: protos/sources/vega/commands/v1/commands.proto [hunk 1]
```diff
@@ -303,3 +303,48 @@ message IssueSignatures {
...
 }
+
+// Request the creation of a team.
+//
+// The creator of the team automatically becomes
+// the team leader, called referrer. The referrer cannot be changed. The creation
+// is possible if the creator does not hold any liquidity provision, and is not
+// already a team member (referrer nor referee).
+// When the party becomes a referrer, he cannot submit liquidity provision
+// anymore.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:314]:
```suggestion
// Request for creating a referral team
//
// Team creator automatically becomes the team leader,
// called a referrer. The referrer cannot be changed. Creating a team
// is possible if the creator does not provide liquidity, and is not
// already a team member - referrer or referee.
// When a party becomes a referrer, they can no longer submit liquidity provisions.
```

File: protos/sources/vega/commands/v1/commands.proto [hunk 2]
```diff
@@ -303,3 +303,48 @@ message IssueSignatures {
...
+message CreateTeam {
+  // Specifies whether the team should be eligible for team based rewards.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:316]:
```suggestion
  // Specifies whether the team is eligible for team based rewards.
```

File: protos/sources/vega/commands/v1/commands.proto [hunk 3]
```diff
@@ -303,3 +303,48 @@ message IssueSignatures {
...
+  bool enable_rewards = 1;
+  // Optional team name to be added to the referral banner.
+  optional string name = 2;
+  // Optional link to a team forum, discord, etc.
+  optional string team_url = 3;
+  // Optional link to an image to be used as the team avatar.
+  optional string avatar_url = 4;
+}
+
+// Request the update of a team properties.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:326]:
```suggestion
// Request for updating a team's properties
```

File: protos/sources/vega/commands/v1/commands.proto [hunk 4]
```diff
@@ -303,3 +303,48 @@ message IssueSignatures {
...
+message UpdateTeam {
+  // Team ID of the team to update.
+  string team_id = 1;
+  // Specifies whether the team should be eligible for team based rewards.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:330]:
```suggestion
  // Specifies whether the team is eligible for team based rewards.
```

File: protos/sources/vega/commands/v1/commands.proto [hunk 5]
```diff
@@ -303,3 +303,48 @@ message IssueSignatures {
...
+  bool enable_rewards = 2;
+  // Optional team name to be added to the referral banner.
+  optional string name = 3;
+  // Optional link to a team forum, discord, etc.
+  optional string team_url = 4;
+  // Optional link to an image to be used as the team avatar.
+  optional string avatar_url = 5;
+}
+
+// Request from the party to join a team.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:340]:
```suggestion
// Request to join a team
```

File: protos/sources/vega/commands/v1/commands.proto [hunk 6]
```diff
@@ -303,3 +303,48 @@ message IssueSignatures {
...
+//
+// A party that joins a team is called a referee. Joining is possible if the
+// joiner does not hold any liquidity provision. A referee cannot create a
+// team, and cannot join multiple teams at once. He will have to switch team.
+// To switch team, the referee can ask to join another team, and the switch will
+// be effective at the end of the epoch.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:346]:
```suggestion
// A party that joins a referral team is called a referee. A referee cannot 
// be a liquidity provider, nor can they create a team or join multiple teams.
// To switch teams, the referee can ask to join another team, and the switch will
// be effective at the end of the epoch.
```

**Review by jeremyletang — APPROVED** [review]

**Review by gordsport — APPROVED** [review]
LGTM

**Pull request merged**